src/openfermion/measurements/_qubit_partitioning_test.py

```diff
@@ -0,0 +1,181 @@
+                print(i, j)
+                flag = False
+                bpi = binary_partition_iterator(qubit_list)
+                for p1, p2 in bpi:
+                    print(p1, p2)
+                    if (i in p1 and j in p2) or (j in p1 and i in p2):
+                        flag = True
+                self.assertTrue(flag)
+
+    def test_exception(self):
+        with self.assertRaises(ValueError):
+            bpi = binary_partition_iterator([])
+            next(bpi)
+
+    def test_partition_of_two(self):
+        bpi = binary_partition_iterator([0, 1])
+        count = 0
+        for p1, p2 in bpi:
+            count += 1
+            self.assertEqual(p1[0], 0)
+            self.assertEqual(p2[0], 1)
+        self.assertEqual(count, 1)
+
+    def test_zero_counting(self):
+        bpi = binary_partition_iterator([0, 1], 0)
+        with self.assertRaises(StopIteration):
+            next(bpi)
+
+
+class PartitionIteratorTest(unittest.TestCase):
+
+    def test_unary_case(self):
+        qubit_list = list(range(6))
+        bpi = partition_iterator(qubit_list, 1)
+        for p1, in bpi:
+            self.assertEqual(p1, qubit_list)
+
+    def test_binary_case(self):
+        qubit_list = list(range(6))
+        for i in range(6):
+            for j in range(i+1, 6):
+                print(i, j)
+                flag = False
+                bpi = partition_iterator(qubit_list, 2)
+                for p1, p2 in bpi:
+                    print(p1, p2)
+                    if (i in p1 and j in p2) or (j in p1 and i in p2):
+                        flag = True
+                self.assertTrue(flag)
+
+    def test_exception(self):
+        with self.assertRaises(ValueError):
+            pi = partition_iterator([1, 2], 3)
+            next(pi)
+
+    def test_threepartition_three(self):
+        bpi = partition_iterator([1, 2, 3], 3)
+        count = 0
+        for p1, p2, p3 in bpi:
+            print(p1, p2, p3)
+            self.assertEqual(len(p1), 1)
+            self.assertEqual(p1[0], 1)
+            self.assertEqual(len(p2), 1)
+            self.assertEqual(p2[0], 2)
+            self.assertEqual(len(p3), 1)
+            self.assertEqual(p3[0], 3)
+            count += 1
+        self.assertEqual(count, 1)
+
+    def test_partition_three(self):
+        for num_qubits in range(1, 16):
+            qubit_list = list(range(num_qubits))
+            for i in range(num_qubits):
+                for j in range(i+1, num_qubits):
+                    for k in range(j+1, num_qubits):
+                        print('Testing {}, {}, {}'.format(i, j, k))
+                        pi = partition_iterator(qubit_list, 3)
+                        count = 0
+                        for p1, p2, p3 in pi:
+                            print('Partition obtained: ', p1, p2, p3)
+                            if max([sum([1 for x in p if x in [i, j, k]])
+                                    for p in [p1, p2, p3]]) == 1:
```

Comment:
The brackets `[` are unnecessary in the arguments to `max` and `sum`.

Reply:
Thanks, I didn't know that